ui/app/routes/jobs/run/templates/index.js

```diff
@@ -5,21 +5,28 @@ export default class RunTemplatesRoute extends Route {
   @service can;
   @service store;
 
-  beforeModel() {
-    if (
-      this.can.cannot('write variable', null, {
-        namespace: '*',
-        path: '*',
-      })
-    ) {
+  beforeModel(transition) {
+    const hasPermissions = this.can.can('write variable', null, {
+      namespace: '*',
+      path: '*',
+    });
+
+    // We create a job with no id in jobs.run that is populated by this form.
+    // A user cannot start at this route.
+    const wasJobModelCreated = transition.from?.name === 'jobs.run.index';
```

Comment:
`note`: Previously the `jobs.run` route created an empty `job` without an `id` upon entering the route and there were two behaviors:

1. The page acted like a form, that upon hitting `Plan` it would set the `id`, `namespace` and `plainId` and the newly created `job`.
2. Upon exiting the page, it would destroy the `model`.

Now, we're creating a multi-page form there's more nuance here. 

Challenge:  We're creating a `job` without an `id` or any queryable attributes and then we navigate to a brand new page that loses the context of the `route`.

- How do we preserve this state?
- How do we ensure proper clean-up and maintenance of our `ember-data` store?

These are architectural questions of this sub-route and not addressed in this PR. The simple solution would be to recognize that we're in a multi-step form and only allow the user to enter the form via `jobs.run` (the beginning of the form). We currently don't support page refresh behavior to preserve draft state in our current version of the `JobEditor` (this is a pretty frustrating challenge that would be worth solving while we are here).

Nomad currently has 4 forms and this is our first multi-page form.

I think we want to think about this problem architecturally and consider creating a `FormService` that would be responsible for creating/updating the new entity because trying to manage this across the 4 pages in this view could make it challenging for us to keep track of permissioning and the non-queryable `job` that we're in the process of creating.

I'd propose having a separate PR to handle this.

We totally could just keep this logic within our `routes` and use Ember API's to help us here which would be a little faster in the short-term, however, it could become a big ball of mud.